users/api/http/endpoint.go

```diff
@@ -27,6 +28,54 @@ func registrationEndpoint(svc users.Service) endpoint.Endpoint {
 	}
 }
 
+// Password reset endpoint serves post request with email of the user
+// for whom password reset flow is to be initiated.
+// If request is succsesfull email with reset link will be sent to the
+// email specified in the request.
+// Link contains token that has TTL that needs to be verified.
+// When user gets email with reset password link this endpoint can serve
+// that request and return response into the ui form which than can
+// be used to enter email and new password and then request can be submited
+// to the post endpoint which actually changes password.
+func passwordResetRequestEndpoint(svc users.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(passwResetReq)
+		res := resetPassRes{}
+		email := req.Email
+		tok, err := svc.GenerateResetToken(ctx, email)
+		if err != nil {
+			res.Error = err.Error()
+			return res, err
+		}
+
+		err = svc.SaveToken(ctx, email, tok)
+		if err != nil {
+			res.Error = err.Error()
+			return res, nil
+		}
+		token.SendToken(req.Host, email, tok)
```

Comment:
I think that all of these methods should be under one method in the `Service` interface.

Reply:
Looks like I wasn't clear enough. All svc methods that are called here should be reduced to one method call of the `Service` interface. This `token` should not be present here but in the `Service` implementation. It's ok that it's a separate method, but here we should see only one method call to the `svc`. Point is that `http` layer depends only on the `users.Service`, while service itself will then call multiple other methods (it should contain business logic).